src/radiation/radiation.cpp

```diff
@@ -304,27 +305,115 @@ std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
     namespace c = radmoment_cons;
     namespace i = radmoment_internal;
 
+    std::string recon = pin->GetOrAddString("radiation", "recon", "linear");
+    PhoebusReconstruction::ReconType rt = PhoebusReconstruction::ReconType::linear;
+    if (recon == "weno5" || recon == "weno5z") {
+      PARTHENON_REQUIRE_THROWS(parthenon::Globals::nghost >= 4,
+                               "weno5 requires 4+ ghost cells");
+      rt = PhoebusReconstruction::ReconType::weno5z;
+    } else if (recon == "mp5") {
+      PARTHENON_REQUIRE_THROWS(parthenon::Globals::nghost >= 4,
+                               "mp5 requires 4+ ghost cells");
+      if (cfl > 0.4) {
+        PARTHENON_WARN("mp5 often requires smaller cfl numbers for stability");
+      }
+      rt = PhoebusReconstruction::ReconType::mp5;
+    } else if (recon == "linear") {
+      rt = PhoebusReconstruction::ReconType::linear;
+    } else if (recon == "constant") {
+      rt = PhoebusReconstruction::ReconType::constant;
+    } else {
+      PARTHENON_THROW("Invalid Reconstruction option.  Choose from "
+                      "[constant,linear,mp5,weno5,weno5z]");
+    }
+    params.Add("Recon", rt);
+
+    const std::string recon_fixup_strategy_str =
+        pin->GetOrAddString("radiation", "recon_fixup_strategy", "bounds");
+    ReconFixupStrategy recon_fixup_strategy;
+    if (recon_fixup_strategy_str == "none") {
+      recon_fixup_strategy = ReconFixupStrategy::none;
+    } else if (recon_fixup_strategy_str == "bounds") {
+      recon_fixup_strategy = ReconFixupStrategy::bounds;
+    } else {
+      PARTHENON_FAIL("\"radiation/recon_fixup_strategy\" option unrecognized!");
+    }
+    params.Add("recon_fixup_strategy", recon_fixup_strategy);
+
+    //    std::string solver = pin->GetOrAddString("radiation", "riemann", "hll");
+    //    riemann::solver rs = riemann::solver::HLL;
+    //    if (solver == "llf") {
+    //      rs = riemann::solver::LLF;
+    //    } else if (solver == "hll") {
+    //      rs = riemann::solver::HLL;
+    //    } else {
+    //      PARTHENON_THROW("Invalid Riemann Solver option. Choose from [llf, hll]");
+    //    }
+    //    params.Add("RiemannSolver", rs);
+
+    const std::string src_solver_name =
+        pin->GetOrAddString("radiation", "src_solver", "fourd");
+    SourceSolver src_solver;
+    if (src_solver_name == "zerod") {
+      src_solver = SourceSolver::zerod;
+    } else if (src_solver_name == "oned") {
+      src_solver = SourceSolver::oned;
+    } else if (src_solver_name == "fourd") {
+      src_solver = SourceSolver::fourd;
+    } else {
```

Comment:
will we always need all these?

Reply:
Long term we will at *least* need `oned` and a new `twod` (that solves for both `T` and `Ye`). Let me think about whether it's better to just remove `fourd` for now (and revert later if we need it again) -- it would certainly clean up the code a lot and make implementing `twod` a bit less painful.